### DISCUSSION TITLE
Enabling SSL causes connection to close randomly

### DISCUSSION THREAD

**Discussion by ethan-tonic** [Kludex/uvicorn#2133] [Potential Issue]
I was setting up uvicorn with fastapi and enabled SSL. After enabling SSL, I noticed every now and then when I send a request, I will get the following error `ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host`. It seems to be a Windows specific error, but I couldn't find any solutions to it online.

### How to reproduce
**This only occurs on Windows**
```
--- main.py
import uvicorn
from fastapi import FastAPI
from fastapi.staticfiles import StaticFiles

from api.api import api_router

api_str = "/api/v1"
app = FastAPI(title="test", openapi_url=f"{api_str}/openapi.json")

app.include_router(api_router, prefix=api_str)

if __name__ == "__main__":
        uvicorn.run(
            "main:app",
            port=8001,
            host="0.0.0.0",
            ssl_keyfile="./key.pem",
            ssl_certfile="./cert.pem",
        )

--- api/api.py
from fastapi import APIRouter

from api.endpoints import test_endpoint

api_router = APIRouter()
api_router.include_router(test_endpoint.router, prefix="/test", tags=["test"])

--- api/endpoints/test_endpoint.py
from typing import Any
from fastapi import APIRouter

router = APIRouter()

@router.get("/")
async def root() -> Any:
    return {"message": "Hello World"}
```

To generate the certificate and key. I did the following command `openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 10000 -nodes`

After doing all the above steps, I just would randomly request the api test endpoint and while it works most of the time, every now and then it will fail with the following error.

```
ERROR [asyncio] Exception in callback _ProactorBasePipeTransport._call_connection_lost(None)
handle: <Handle _ProactorBasePipeTransport._call_connection_lost(None)>
Traceback (most recent call last):
  File "C:\Users\ethantonic\AppData\Local\Programs\Python\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\ethantonic\AppData\Local\Programs\Python\Python311\Lib\asyncio\proactor_events.py", line 165, in _call_connection_lost
    self._sock.shutdown(socket.SHUT_RDWR)
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
```

If it helps, I am on version 3.11.6 of Python and on Windows 11 for my OS. My uvicorn version is 0.23.2.

**Comment by ramjke** [comment]
Did you find a solution? I’m facing the same rare issue with my FastAPI behind Cloudflare and a Windows client

**Reply by amias-channer** [reply]
seeing the same on windows 11 , i made my certs with mkcert.  using port 8000   Same python 3.11 , running both via _main_ and launched by uvicorn .  Adding SSL brought about these errors.   These connections its getting worried about don't seem to be from my client , i wondered if Defender or something similar was poking about so i changed the port to some random numbers and its still happening. So it seems to be something internal.   The error comes from a big event loop within FastAPI  so its hard to trace but i think ultimately its just something that happens and probably shouldn't fire an error, i can't see any bad effects from it

**Comment by amias-channer** [comment]
seems to be this issue with the asyncio module using a different event loop its not used to https://stackoverflow.com/questions/62412754/python-asyncio-errors-oserror-winerror-6-the-handle-is-invalid-and-runtim
